backend/quickTA/users/views.py

```diff
@@ -184,6 +185,9 @@ def get(self, request):
         
         if user_id: user = get_object_or_404(User, user_id=user_id)
         else: user = get_object_or_404(User, utorid=utorid)
+
+        if user.user_role == 'ST':
+            return JsonResponse([course for course in user.courses if course.end_date > datetime.now()], safe=False)
```

Comment:
user.courses stores only course_ids --> need to acquire the Course object to access its end_date attribute